zilencer/views.py

```diff
@@ -114,22 +116,47 @@ def deactivate_remote_server(
     return json_success(request)
 
 
+# The Pydantic.StringConstraints does not have validation for the string to be
+# of the specified length. So, we need to create a custom validator for that.
+
+
+def check_string_fixed_length(string: str, length: int) -> Optional[str]:
+    if len(string) != length:
+        raise PydanticCustomError(
+            "string_fixed_length",
+            "",
+            {
+                "length": length,
+            },
+        )
+    return string
+
+
 @csrf_exempt
 @require_post
-@has_request_variables
+@typed_endpoint
 def register_remote_server(
     request: HttpRequest,
-    zulip_org_id: str = REQ(str_validator=check_string_fixed_length(RemoteZulipServer.UUID_LENGTH)),
-    zulip_org_key: str = REQ(
-        str_validator=check_string_fixed_length(RemoteZulipServer.API_KEY_LENGTH)
-    ),
-    hostname: str = REQ(str_validator=check_capped_string(RemoteZulipServer.HOSTNAME_MAX_LENGTH)),
-    contact_email: str = REQ(),
-    new_org_key: Optional[str] = REQ(
-        str_validator=check_string_fixed_length(RemoteZulipServer.API_KEY_LENGTH), default=None
-    ),
+    *,
+    zulip_org_id: Annotated[
+        str,
+        RequiredStringConstraint,
+        AfterValidator(lambda s: check_string_fixed_length(s, RemoteZulipServer.UUID_LENGTH)),
+    ],
```

Comment:
This one might also be appropriate to just redeclare as a UUID validator?

Reply:
It's also being used to validate `zulip_org_key`, mostly string related stuff

Reply:
Sorry, I don't mean that `check_string_fixed_length` should be a UUID validator, but that probably this parameter could be validated using the built-in Pydantic UUID validator, rather than just validating its length as a string.